### PR TITLE
Add new file enter name validation change

### LINKED ISSUE #129: Limit the New contract name input to 16 characters
## Environment/Browser
Lab 1.0.0

## Description
Input field (name) in _New contract_ doesn't enforce maximum characters limit.

## Steps to reproduce
1. Given an open project
2. Create a new contract file by clicking the _New contract_ button
3. Observe the maximum allowed input length in _Please give the contract a name_ alert box

## Expected result
Limit _Please give the contract a name_ input field to 16 characters.

## Actual result
There is apparently no limitation at the moment. The check happens afterwards, leading into a repetition that could have otherwise been prevented.

### PR DESCRIPTION
**Pull request by waheed25**
this closes #129

### CODE REVIEW

File: src/components/projecteditor/control/control.js [hunk 1]
```diff
@@ -860,13 +860,35 @@ export default class Control extends Component {
...
+            else if(!name.match(/^([a-zA-Z0-9-_]+)$/)) {
+                name = prompt('Illegal contract name. Only A-Za-z0-9, dash (-) and underscore (_) allowed.',name);
+            }
+            else {
+                alert('Something strange is happened please report this issue')
+            }
+
         }
+        while (!name || name.length > 16);
+        // if(name.length>16){
```
Comment (filippsen) [inline src/components/projecteditor/control/control.js]: Please do not add unused code or commented out code, unless it serves as documentation or guidance

File: src/components/projecteditor/control/control.js [hunk 2]
```diff
@@ -860,13 +860,35 @@ export default class Control extends Component {
...
-        var name;
-        name = prompt("Please give the contract a name:");
-        if (!name) return;
-        if(!name.match(/^([a-zA-Z0-9-_]+)$/) || name.length > 16) {
-            alert('Illegal contract name. Only A-Za-z0-9, dash (-) and underscore (_) allowed. Max 16 characters.');
-            return;
+        var name = undefined;
+        do {
+            if(name === undefined ){
+                name = prompt('Please give the contract a name:', "Only 16 character Allowed");
```
Comment (filippsen) [inline src/components/projecteditor/control/control.js]: Change the prompt to keep the same writing style as before:
```
name = prompt('Please give the contract a name (up to 16 characters):');
```

File: src/components/projecteditor/control/control.js [hunk 3]
```diff
@@ -860,13 +860,35 @@ export default class Control extends Component {
...
+            }
+            else if (name !== null && name.length > 16) {
+                name = prompt('Only 16 character Allowed:', name);
```
Comment (filippsen) [inline src/components/projecteditor/control/control.js]: Change the prompt to keep the same writing style as before:
```
name = prompt('Please give the contract a name (up to 16 characters):');
```

File: src/components/projecteditor/control/control.js [hunk 4]
```diff
@@ -860,13 +860,35 @@ export default class Control extends Component {
...
+            }
+            else if(!name){
+                return null;
+            }
+            else if(!name.match(/^([a-zA-Z0-9-_]+)$/)) {
```
Comment (filippsen) [inline src/components/projecteditor/control/control.js]: This condition likely never gets executed, considering `name` can be set in the very first `if`, rendering the loop condition `false` for the next iteration.

Comment (waheed25) [reply src/components/projecteditor/control/control.js]: yeah, there was a problem that is fixed now.

File: src/components/projecteditor/control/control.js (hunk 1 above)
Comment (waheed25) [reply src/components/projecteditor/control/control.js]: okay, I will.

**Comment by filippsen** [comment]
Issue #129 has been labelled _wontfix_

**Pull request closed without merging**